### PR TITLE
Remove unneeded C APIs + some refactoring.

### PR DESCRIPTION
**Pull request by pranavsharma**
**Description**: Remove unneeded C APIs + some refactoring.

**Motivation and Context**
We want to keep the C API to the bare minimum.

### CODE REVIEW

**Comment by snnn** [comment]
At this time point, we won't provide serialization support for ortvalues.  If a customer has such a need, he can have his own serialization mechanism.

File: onnxruntime/test/onnx/TestCase.cc [hunk 1]
```diff
@@ -16,6 +14,8 @@
 #include <map>
 #include <regex>
 #include "OrtValueList.h"
+#undef OPTIONAL  // to get around onnxruntime\cmake\external\onnx\onnx/common/constants.h(14): error C2513: 'bool': no variable declared before '='
```
Comment (snnn) [inline onnxruntime/test/onnx/TestCase.cc]: You may also workaround it by reordering the header files.

Comment (pranavsharma) [reply onnxruntime/test/onnx/TestCase.cc]: fixed.

**Comment by pranavsharma** [comment]
@snn can this be approved?

**Pull request merged**